.github/workflows/publish-vscode-extension.yaml

```diff
@@ -0,0 +1,45 @@
+name: Publish VSCode extension to Marketplace and OpenVSX
+
+on:
+  # Only allow this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  NODE_OPTIONS: '--max_old_space_size=4096'
+
+jobs:
+  publish-vscode-extension:
+    name: 'Publish VSCode extension to Marketplace and OpenVSX'
+    environment: publish-vscode-extension
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+
+      - name: Setup antlr4
+        uses: ./.github/actions/setup-antlr4
```

Comment:
feels like it's soon time for us to consider caching builds in CI, so we only do one build per merge and just download the artifacts